app/src/main/java/org/alexcode/snake/Settings.java

```diff
@@ -0,0 +1,119 @@
+package org.alexcode.snake;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.CompoundButton;
+import android.widget.Spinner;
+import android.widget.Switch;
+
+public class Settings extends AppCompatActivity implements AdapterView.OnItemSelectedListener {
+    private Switch musicSwitch, fxSoundSwitch;
+    private Spinner languageSpinner;
+    private String music, fxSounds, language;
+    private Button saveSettings;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_settings);
+        //get music and fxSounds values for local device
+        music = PlayerPreferences.getMusic();
+        fxSounds = PlayerPreferences.getFxSounds();
+        language = PlayerPreferences.getGameLanguage();
+        //initialize the elements
+        initPlayerSettings();
+        //set switchers
+        setSwitchers();
+        // set dropdown language list
+        setDropDownLanguageList();
+        //set save button
+        saveSettings.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent intent =  new Intent(Settings.this, MainActivity.class);
+                startActivity(intent);
+            }
+        });
+    }
+
+    private void initPlayerSettings() {
+        musicSwitch = findViewById(R.id.switchMusic);
+        fxSoundSwitch = findViewById(R.id.switchSounds);
+        languageSpinner = findViewById(R.id.spinnerLanguageSelector);
+        saveSettings = findViewById(R.id.saveSettings);
+    }
+
+    private void setDropDownLanguageList() {
+        //set values for dropdown languages list
+        ArrayAdapter<CharSequence> languageAdapter = ArrayAdapter.createFromResource(Settings.this, R.array.available_languages, android.R.layout.simple_spinner_item);
+        languageAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        languageSpinner.setAdapter(languageAdapter);
+        languageSpinner.setOnItemSelectedListener(Settings.this);
+        switch (language) {
+            case "English":
+                languageSpinner.setSelection(0);
+                break;
+            case "Spanish":
+                languageSpinner.setSelection(1);
+                break;
+            case "German":
+                languageSpinner.setSelection(2);
+                break;
+            default:
+                languageSpinner.setSelection(3);
+                break;
+        }
```

Comment:
isn't there a package that ensures all need for the language part? because the language part has to be something very usual in mobile development

Reply:
You can get the language of the device but you need to make the translations. https://developer.android.com/training/basics/supporting-devices/languages 
I choose to let the user set the language for the application no matter what language settings have the device.